src/Websites/Websites/Cmdlets/WebApps/PublishAzureWebApp.cs

```diff
@@ -0,0 +1,107 @@
+        private const int NumStatusChecks = 600;
+
+        [Parameter(Mandatory = true, HelpMessage = "The path of the archive file. ZIP, WAR, and JAR are supported.")]
+        [ValidateNotNullOrEmpty]
+        public string ArchivePath { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Deploy the web app without prompting for confirmation.")]
+        public SwitchParameter Force { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+            User user = WebsitesClient.GetPublishingCredentials(ResourceGroupName, Name, Slot);
+
+            HttpResponseMessage r;
+            string deployUrl;
+            string deploymentStatusUrl = user.ScmUri + "/api/deployments/latest";
+
+            if (ArchivePath.ToLower().EndsWith("war"))
+            {
+                deployUrl = user.ScmUri + "/api/wardeploy?isAsync=true";
+            }
+            else if (ArchivePath.ToLower().EndsWith("zip") || ArchivePath.ToLower().EndsWith("jar"))
+            {
+                deployUrl = user.ScmUri + "/api/zipdeploy?isAsync=true";
+            }
+            else
+            {
+                throw new Exception("Unknown archive type.");
+            }
+
+            Action zipDeployAction = () =>
+            {
+                using (var s = File.OpenRead(ArchivePath))
+                {
+                    HttpClient client = new HttpClient();
+                    var byteArray = Encoding.ASCII.GetBytes(user.PublishingUserName + ":" + user.PublishingPassword);
+                    client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", Convert.ToBase64String(byteArray));
+                    HttpContent fileContent = new StreamContent(s);
+                    fileContent.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("multipart/form-data");
+                    r = client.PostAsync(deployUrl, fileContent).Result;
+
+                    int numChecks = 0;
+                    do
+                    {
+                        Thread.Sleep(TimeSpan.FromSeconds(2));
+                        r = client.GetAsync(deploymentStatusUrl).Result;
+                        numChecks++;
+                    } while (r.StatusCode == HttpStatusCode.Accepted && numChecks < NumStatusChecks);
+
+                    if (r.StatusCode == HttpStatusCode.Accepted && numChecks >= NumStatusChecks)
+                    {
+                        WriteWarning("Maximum status polling time exceeded. Deployment is still in progress.");
+                    }
+                    else if (r.StatusCode != HttpStatusCode.OK)
+                    {
+                        WriteWarning("Deployment failed with status code=" + r.StatusCode + " reason=" + r.ReasonPhrase);
```

Comment:
why do you write  to Warning stream (not Error)?

Reply:
Wasn't sure what the proper use of warning vs error would be. I've changed it to error now.